Sources/Utility/Process.swift

```diff
@@ -0,0 +1,253 @@
+        ///
+        case error
+    }
+
+    /// Typealias for process id type.
+    public typealias ProcessID = pid_t
+
+    /// The arguments to execute.
+    public let arguments: [String]
+
+    /// The enviornment with which the process was executed.
+    public let environment: [String: String]
+
+    /// The return code of the process, available once process has finished execution.
+    public private(set) var returnCode: Int32 = 0
+
+    /// The process id of the spawned process, available after the process is launched.
+    public private(set) var processID = ProcessID()
+
+    /// The output of the process, available after execution only if process was asked to save the output.
+    public private(set) var output = ""
+
+    /// If the subprocess has launched.
+    public private(set) var launched = false
+
+    /// The pipes to redirect and save the output of the subprocess.
+    private var saveOutputPipe: [Int32] = [0, 0]
+
+    /// If the process output should be redirected and saved.
+    public var saveOutput = false
+
+    /// Create a new process instance.
+    ///
+    /// - Parameters:
+    ///   - arguments: The arugments for the subprocess.
+    ///   - environment: The environment to pass to subprocess. By default the current process environment
+    ///     will be inherited.
+    public init(arguments: [String], environment: [String: String] = env()) {
+        self.arguments = arguments
+        self.environment = environment
+    }
+
+    /// Launch the subprocess.
+    ///
+    /// - Throws: Process.Error
+    public func launch() throws {
+        assert(!launched, "It is not allowed to launch the same process object again.")
+        launched = true
+
+        // Initialize the spawn attributes.
+        var attributes: posix_spawnattr_t? = nil
+        posix_spawnattr_init(&attributes)
+
+        // Unmask all signals.
+        var noSignals = sigset_t()
+        sigemptyset(&noSignals)
+        posix_spawnattr_setsigmask(&attributes, &noSignals)
+
+        // Reset all signals to default behavior.
+        var mostSignals = sigset_t()
+        sigfillset(&mostSignals)
+        sigdelset(&mostSignals, SIGKILL)
+        sigdelset(&mostSignals, SIGSTOP)
+        posix_spawnattr_setsigdefault(&attributes, &mostSignals)
+
+        // Establish a separate process group.
+        posix_spawnattr_setpgroup(&attributes, 0)
+
+        // Set the attribute flags.
+        var flags = POSIX_SPAWN_SETSIGMASK | POSIX_SPAWN_SETSIGDEF
+        flags |= POSIX_SPAWN_SETPGROUP
+        flags |= POSIX_SPAWN_CLOEXEC_DEFAULT
+        posix_spawnattr_setflags(&attributes, Int16(flags))
+
+        #if os(macOS)
+            var fileActions: posix_spawn_file_actions_t? = nil
+        #else
+            var fileActions = posix_spawn_file_actions_t()
+        #endif
+        posix_spawn_file_actions_init(&fileActions)
+
+        // Workaround for https://sourceware.org/git/gitweb.cgi?p=glibc.git;h=89e435f3559c53084498e9baad22172b64429362
```

Comment:
Ah here it is: https://github.com/apple/swift-package-manager/pull/680
Doesn't look like we need to `#if`